src/main/java/com/softserve/kickscootertrip/service/TripService.java

```diff
@@ -0,0 +1,89 @@
+package com.softserve.kickscootertrip.service;
+
+import com.softserve.kickscootertrip.model.dto.TripDto;
+import com.softserve.kickscootertrip.model.dto.UserDto;
+import com.softserve.kickscootertrip.model.entity.TripEntity;
+import com.softserve.kickscootertrip.repository.TripRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.core.convert.ConversionService;
+import org.springframework.kafka.annotation.KafkaListener;
+import org.springframework.kafka.core.KafkaTemplate;
+import org.springframework.stereotype.Service;
+
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.time.temporal.ChronoUnit;
+import java.util.Date;
+import java.util.List;
+import java.util.UUID;
+
+@Service
+@RequiredArgsConstructor
+public class TripService {
+
+    private static final double RADIUS_EARTH = 6371008.67;
```

Comment:
👀

Reply:
I don't think the radius is tightly related to the trips. It seems like it should be in a separate service.